explore/src/main/scala/explore/targeteditor/SearchForm.scala

```diff
@@ -55,22 +54,19 @@ object SearchForm:
       .useEffectWithDepsBy((props, _, _, _, _) => props.targetName.get)((_, _, term, enabled, _) =>
         name => term.set(name) >> enabled.setState(true)
       )
-      .useEffectWithDepsBy((props, _, _, _, _) => props.id)((props, _, _, _, _) =>
-        _ => // Auto-select name field on new targets.
-          // Accessing dom elements by id is not ideal in React, but passing a ref to the <input> element of a
-          // Form.Input is a bit convoluted. We should reevaluate when and if we switch to another component library.
-          // TODO REEVALUATE NOW!
-          CallbackTo(
-            Option(dom.document.getElementById("search"))
-              .foldMap(node =>
-                Callback(timers.setTimeout(0)(node.asInstanceOf[dom.HTMLInputElement].select()))
+      .useRefToVdom[dom.HTMLButtonElement]
+      .useRefToVdom[dom.HTMLInputElement]
+      .useEffectWithDepsBy((props, _, _, _, _, _, inputRef) => (props.id, inputRef))(
+        (props, _, _, _, _, _, _) =>
+          // Auto select name field on new targets
+          (_, inputRef) =>
+            inputRef.get
+              .flatMap(
+                _.foldMap(i => Callback(timers.setTimeout(50)(i.select())))
```

Comment:
With a timeout of zero, this solution had about the same reliability as the previous one. That is to say, the input always got selected but sometimes the cursor was at the end of the field rather than selecting all of the text. With the 50ms delay it has worked every time I tried it.

Reply:
Can this be done as `Callback.delayMs(50)`?

Reply:
I'm not sure. `delayMs` returns an `AsyncCallback` which seemed harder to deal with. I'd have to turn it back into a `Callback`.